File: TemplateVB/TheForm.vb
Imports System.Windows.Forms
Imports Visio = Microsoft.Office.Interop.Visio

Public Class TheForm

    Private ReadOnly _window As Visio.Window

    Public Sub New(window As Visio.Window)
        _window = window
        InitializeComponent()
    End Sub

    Private Sub button1_Click(sender As Object, e As EventArgs) Handles button1.Click
        MessageBox.Show(_window.Document.Name)
    End Sub

End Class

File: TemplateCS/Addin/TheForm.cs
using System.Windows.Forms;
using Visio = Microsoft.Office.Interop.Visio;

namespace $csprojectname$
{
    public partial class TheForm : Form
    {
        private readonly Visio.Window _window;

        /// <summary>
        /// Form constructor, receives parent Visio diagram window
        /// </summary>
        /// <param name="window">Parent Visio diagram window</param>
        public TheForm(Visio.Window window)
        {
            _window = window;
            InitializeComponent();
        }

        /// <summary>
        /// Sample method. We just show a Message Box. 
        /// Do something meaningful here instead.
        /// </summary>
        private void button1_Click(object sender, System.EventArgs e)
        {
            MessageBox.Show(_window.Document.Name);
        }
    }
}
